### DISCUSSION TITLE
input label class

### DISCUSSION THREAD

**Discussion by behvandi11** [wireui/wireui#296] [Ideas]
Hi, is there any way to use dynamic class for input label? if use class in tag: <x-input class="require"> this class use in input. this use case in very good but it would be great if we could send props for the label as well

**Comment by ph7jack** [comment]
Could you explain more? What are the benefits to pass classes to the label? How do you intend to use it?

**Reply by behvandi11** [reply]
Thanks for the reply. For example, if I can pass the 'require' class to the label then I can define a style that displays a red star after the label for required fields, and some things like that.

**Reply by trippo** [reply]
I think is useful also to have a required param to add the * on the label

**Comment by behvandi11** [comment]
@PH7-Jack Hi Pedro. I have created some views overridden for show a red star after the label. the red star displays just if we write a required attribute(standard html attribute) in the x-component element. if you like this feature, I can move these code changes to the package core and send a pull request for merge

star demo:
![InkedUrlaube](https://user-images.githubusercontent.com/75178945/186202211-16017dc0-5681-41d2-bf3e-dcf24c366609.jpg)

code:
![req](https://user-images.githubusercontent.com/75178945/186202408-b63f417f-f189-4e28-8181-deeafd1e327c.JPG)

**Reply by ph7jack** [reply]
@behvandi11 It's interesting, if possible, create a PR, I'm glad about all contributions. Make it a config (default false) on the wireui config file

**Reply by trippo** [reply]
I think, after implementation of the tooltip , it will be useful to add also an info icon with a description/help of the input

**Reply by blockhunts** [reply]
@PH7-Jack  Is this feature available now?

**Reply by Akhmami** [reply]
@ph7jack Hope this feature available.

**Reply by behvandi11** [reply]
@Akhmami I do this by css trick. wrap your input by a div and give required class to the div then in global css add:
```
.required label.block:after{
    content: "*";
    color: red;
    display: inline;
}
```

this trick adds one extra step, but it works.
